pareto/utilities/get_data.py

```diff
@@ -23,19 +23,182 @@
+        "StorageCapacityIncrements",
+        "TreatmentCapacityIncrements",
+        "TreatmentEfficiency",
+        "RemovalEfficiency",
+        "DisposalExpansionCost",
+        "StorageExpansionCost",
+        "TreatmentExpansionCost",
+        "PipelineCapexDistanceBased",
+        "PipelineCapexCapacityBased",
+        "PipelineCapacityIncrements",
+        "PipelineExpansionDistance",
+        "Hydraulics",
+        "Economics",
+        "ExternalWaterQuality",
+        "PadWaterQuality",
+        "StorageInitialWaterQuality",
+        "PadStorageInitialWaterQuality",
+        "DisposalOperatingCapacity",
+        "TreatmentExpansionLeadTime",
+        "DisposalExpansionLeadTime",
+        "StorageExpansionLeadTime",
+        "PipelineExpansionLeadTime_Dist",
+        "PipelineExpansionLeadTime_Capac",
+        "SWDDeep",
+        "SWDAveragePressure",
+        "SWDProxPAWell",
+        "SWDProxInactiveWell",
+        "SWDProxEQ",
+        "SWDProxFault",
+        "SWDProxHpOrLpWell",
+        "SWDRiskFactors",
+        "ComponentPrice",
+        "ComponentTreatment",
+        "MinTreatmentFlow",
+        "MinResidualQuality",
+    ]
+
+
 def _read_data(_fname, _set_list, _parameter_list):
     """
     This methods uses Pandas methods to read from an Excel spreadsheet and output a data frame
     Two data frames are created, one that contains all the Sets: _df_sets, and another one that
     contains all the parameters in raw format: _df_parameters
     """
+    pareto_input_set_tab_names = get_valid_input_set_tab_names()
+    pareto_input_parameter_tab_names = get_valid_input_parameter_tab_names()
+
+    if _set_list is not None:
+        valid_set_tab_names = pareto_input_set_tab_names
+        valid_set_tab_names.extend(_set_list)
+    else:
+        valid_set_tab_names = pareto_input_set_tab_names
+    if _parameter_list is not None:
+        valid_parameter_tab_names = pareto_input_parameter_tab_names
+        valid_parameter_tab_names.extend(_parameter_list)
+    else:
+        valid_parameter_tab_names = pareto_input_parameter_tab_names
+
+    # Check all names available in the input sheet
+    # If the sheet name is unused (not a  valid Set or Parameter tab, not "Overview", and not "Schematic"), raise a warning.
+    unused_tab_list = []
+    df = pd.ExcelFile(_fname)
+    sheet_list = df.sheet_names
+    for name in sheet_list:
+        if (
+            name not in valid_set_tab_names
+            and name not in valid_parameter_tab_names
+            and name != "Overview"
+            and name != "Schematic"
+        ):
+            unused_tab_list.append(name)
+
+    if len(unused_tab_list) > 0:
+        warning_message = (
+            f"Invalid PARETO input has been provided and has NOT been parsed. Check that input tab names match valid PARETO input. If you'd like to read custom tabs, please include as input to get_data(). The following tabs are NOT parsed: "
+            + str(unused_tab_list)
+        )
```

Comment:
I believe the same warning message appears after the results saying that the result excel headers are invalid pareto inputs. Not sure where this function is being called again.

Reply:
Same comment as earlier, so resolving this one!

Reply:
Same comment as other discussion. Resolving this conversation!